### PR TITLE
refactor(core): remove DynamicComponent

### PR DESCRIPTION
**Pull request by tbosch**
BREAKING CHANGE:
A dynamic component is just a component that has no @View annotation…

### CODE REVIEW

File: modules/angular2/src/core/annotations_impl/annotations.js [hunk 1]
```diff
@@ -542,6 +542,51 @@ export class Directive extends Injectable {
...
+ * There is nothing special about `HelloCmp`, which is a regular Angular component. It can also be used in other static
+ * locations.
+ *
+ * ```
+ * @Component({
+ *   selector: 'dynamic-comp'
+ * })
+ * class DynamicComp {
+ *   helloCmp:HelloCmp;
+ *   constructor(loader:DynamicComponentLoader, location:PrivateComponentLocation) {
```
Comment (rkirov) [inline modules/angular2/src/core/annotations_impl/annotations.js]: Looks wrong, I thought we do `ElementRef`

**Comment by rkirov** [comment]
Looks good, but I think you need more changes to make tests pass. Ping me again, if you make a lot of changes otherwise LGTM.

**Comment by tbosch** [comment]
@rkirov PTAL, there was a bug in `compiler.js`. Also added 2 test cases for the error thrown by the compiler now.

File: modules/angular2/src/core/compiler/template_resolver.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ export class TemplateResolver {
         return annotation;
       }
     }
-
-    throw new BaseException(`No template found for ${stringify(component)}`);
```
Comment (rkirov) [inline modules/angular2/src/core/compiler/template_resolver.js]: I would keep the error here and pass the flag in as `allowMissingTemplate`.

**Comment by rkirov** [comment]
LGTM.

File: modules/angular2/src/core/annotations_impl/annotations.js (hunk 1 above)
Comment (pkozlowski-opensource) [inline modules/angular2/src/core/annotations_impl/annotations.js]: It is not clear from this example what the `location` is, what it points to. Even if this is `ElementRef` we should still add one or two sentences on this.

**Pull request merged**